eventuals/do-all.h

```diff
@@ -67,11 +99,68 @@ struct _DoAll {
                               StoppedException()));
                   if (counter_.fetch_sub(1) == 1) {
                     // You're the last eventual so call the continuation.
-                    k_.Start(std::move(values_));
+                    std::optional<std::exception_ptr> exception = GetExceptionIfExists();
+
+                    if (exception) {
+                      try {
+                        std::rethrow_exception(*exception);
+                      } catch (const StoppedException&) {
+                        k_.Stop();
+                      } catch (...) {
+                        k_.Fail(std::current_exception());
+                      }
+                    }
                   }
                 }));
     }
 
+    std::tuple<
+        std::conditional_t<
+            std::is_void_v<typename Eventuals_::template ValueFrom<void>>,
+            std::monostate,
+            typename Eventuals_::template ValueFrom<void>>...>
+    GetTupleOfValues() {
+      return std::apply(
+          [](auto&... value) {
+            ((CHECK(value.valueless_by_exception() == false)), ...);
+            return std::make_tuple(std::get<0>(value)...);
+          },
+          values_);
+    }
+
+    std::optional<std::exception_ptr> GetExceptionIfExists() {
+      std::optional<std::exception_ptr> exception;
+
+      bool stopped = true;
+
+      auto check_value = [&stopped, &exception](auto& value) {
+        if (std::holds_alternative<std::exception_ptr>(value)) {
+          try {
+            std::rethrow_exception(std::get<std::exception_ptr>(value));
+          } catch (const StoppedException&) {
+          } catch (...) {
+            stopped = false;
+            exception.emplace(std::current_exception());
+          }
+        } else {
+          stopped = false;
+        }
+      };
+
+      std::apply(
+          [check_value](auto&... value) {
```

Comment:
```suggestion
          [&check_value](auto&... value) {
```
Seems like you might as well take by reference?